diff --git a/dist/check.js b/dist/check.js
index 2d506595069bb495af834ce2e9985330ef44ecae..7bbaaf380a487f26e95003bf14446533289535f1 100644
--- a/dist/check.js
+++ b/dist/check.js
@@ -9,7 +9,9 @@ var _debug = _interopRequireDefault(require("debug"));
 var _isCoreModule = _interopRequireDefault(require("is-core-module"));
 var _lodash = _interopRequireDefault(require("lodash"));
 var _readdirp = _interopRequireDefault(require("readdirp"));
-var _minimatch = _interopRequireDefault(require("minimatch"));
+var _minimatch = {
+  default: require("minimatch").minimatch
+};
 var _requirePackageName = _interopRequireDefault(require("require-package-name"));
 var _utils = require("./utils");
 var _parser = _interopRequireDefault(require("./utils/parser"));
diff --git a/dist/special/gulp-load-plugins.js b/dist/special/gulp-load-plugins.js
index 71aef8cee1670eb5b91cc2f3718ddeeb43c4a16c..d8c1c37db5d815df3a946e6869e2058a0f4bce8f 100644
--- a/dist/special/gulp-load-plugins.js
+++ b/dist/special/gulp-load-plugins.js
@@ -6,7 +6,9 @@ Object.defineProperty(exports, "__esModule", {
 exports.default = parseGulpPlugins;
 var _path = require("path");
 var _lodash = _interopRequireDefault(require("lodash"));
-var _minimatch = _interopRequireDefault(require("minimatch"));
+var _minimatch = {
+  default: require("minimatch").minimatch
+};
 var _traverse = _interopRequireDefault(require("@babel/traverse"));
 var _es = require("../parser/es7");
 var _importDeclaration = _interopRequireDefault(require("../detector/importDeclaration"));
