diff --git a/expo-dev-launcher-gradle-plugin/src/main/kotlin/expo/modules/devlauncher/DevLauncherPlugin.kt b/expo-dev-launcher-gradle-plugin/src/main/kotlin/expo/modules/devlauncher/DevLauncherPlugin.kt
index b7a856d72f271e5d655d256a2ea2774c6d4356bd..49d90a461f0c7a26c72a71b77009ec92c0e94105 100644
--- a/expo-dev-launcher-gradle-plugin/src/main/kotlin/expo/modules/devlauncher/DevLauncherPlugin.kt
+++ b/expo-dev-launcher-gradle-plugin/src/main/kotlin/expo/modules/devlauncher/DevLauncherPlugin.kt
@@ -32,7 +32,7 @@ abstract class DevLauncherPlugin : Plugin<Project> {
       }
 
       val androidComponents = project.extensions.getByType(AndroidComponentsExtension::class.java)
-      androidComponents.onVariants(androidComponents.selector().withBuildType("debug")) { variant ->
+      androidComponents.onVariants(androidComponents.selector().withBuildType("development")) { variant ->
         variant.instrumentation.transformClassesWith(DevLauncherClassVisitorFactory::class.java, InstrumentationScope.ALL) {
           it.enabled.set(true)
         }
