diff --git a/React/Views/RCTModalHostViewManager.m b/React/Views/RCTModalHostViewManager.m
index b0295e05ae4d54091bd80f77809ca2aeaaa8562b..81f8f4fa738cfe80ec89f32ebe5bab7ed21f5958 100644
--- a/React/Views/RCTModalHostViewManager.m
+++ b/React/Views/RCTModalHostViewManager.m
@@ -75,7 +75,6 @@ - (void)presentModalHostView:(RCTModalHostView *)modalHostView
       modalHostView.onShow(nil);
     }
   };
-  dispatch_async(dispatch_get_main_queue(), ^{
     if (self->_presentationBlock) {
       self->_presentationBlock([modalHostView reactViewController], viewController, animated, completionBlock);
     } else {
@@ -83,7 +82,6 @@ - (void)presentModalHostView:(RCTModalHostView *)modalHostView
                                                         animated:animated
                                                       completion:completionBlock];
     }
-  });
 }
 
 - (void)dismissModalHostView:(RCTModalHostView *)modalHostView
@@ -95,7 +93,6 @@ - (void)dismissModalHostView:(RCTModalHostView *)modalHostView
       [[self.bridge moduleForClass:[RCTModalManager class]] modalDismissed:modalHostView.identifier];
     }
   };
-  dispatch_async(dispatch_get_main_queue(), ^{
     if (self->_dismissalBlock) {
       self->_dismissalBlock([modalHostView reactViewController], viewController, animated, completionBlock);
     } else if (viewController.presentingViewController) {
@@ -106,7 +103,6 @@ - (void)dismissModalHostView:(RCTModalHostView *)modalHostView
       // This, somehow, invalidate the presenting view controller and the modal remains always visible.
       completionBlock();
     }
-  });
 }
 
 - (RCTShadowView *)shadowView
