diff --git a/packages/cds-web-overlays/src/navigation/__stories__/ComposedSystem.stories.mdx b/packages/cds-web-overlays/src/navigation/__stories__/ComposedSystem.stories.mdx
index 0ee92ea16..b0df39d12 100644
--- a/packages/cds-web-overlays/src/navigation/__stories__/ComposedSystem.stories.mdx
+++ b/packages/cds-web-overlays/src/navigation/__stories__/ComposedSystem.stories.mdx
@@ -1,9 +1,9 @@
 import { Meta, Canvas, Story } from '@storybook/addon-docs';
-import { withStoryLabel } from '../../storybook-decorators/withStoryLabel';
+import { withStoryLabel } from '@cbhq/cds-web/storybook-decorators/withStoryLabel';
 
 import { ComposedSystem } from './NavigationStorySetup';
 
-<Meta title="Core Components/Navigation/System" />
+<Meta title="Overlays/Navigation/System" />
 
 # Composed System
 
diff --git a/packages/cds-web-overlays/src/navigation/__stories__/NavLink.stories.mdx b/packages/cds-web-overlays/src/navigation/__stories__/NavLink.stories.mdx
index f308cd12c..8a98b1c43 100644
--- a/packages/cds-web-overlays/src/navigation/__stories__/NavLink.stories.mdx
+++ b/packages/cds-web-overlays/src/navigation/__stories__/NavLink.stories.mdx
@@ -1,10 +1,10 @@
 import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
-import { withStoryLabel } from '../../storybook-decorators/withStoryLabel';
+import { withStoryLabel } from '@cbhq/cds-web/storybook-decorators/withStoryLabel';
 
 import { NavLinkExample } from './NavigationStorySetup';
-import { NavLink } from '../NavLink';
+import { NavLink } from '@cbhq/cds-web/navigation/NavLink';
 
-<Meta title="Core Components/Navigation/NavLink" component={NavLink} />
+<Meta title="Overlays/Navigation/NavLink" component={NavLink} />
 
 # NavLink
 
diff --git a/packages/cds-web-overlays/src/navigation/__stories__/NavigationBar.stories.mdx b/packages/cds-web-overlays/src/navigation/__stories__/NavigationBar.stories.mdx
index cc83e0fb2..3a859fc4a 100644
--- a/packages/cds-web-overlays/src/navigation/__stories__/NavigationBar.stories.mdx
+++ b/packages/cds-web-overlays/src/navigation/__stories__/NavigationBar.stories.mdx
@@ -1,10 +1,10 @@
 import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
-import { withStoryLabel } from '../../storybook-decorators/withStoryLabel';
+import { withStoryLabel } from '@cbhq/cds-web/storybook-decorators/withStoryLabel';
 
 import { NavigationBarFullExample, NavigationBarTitle } from './NavigationStorySetup';
-import { NavigationBar } from '../NavigationBar';
+import { NavigationBar } from '@cbhq/cds-web/navigation/NavigationBar';
 
-<Meta title="Core Components/Navigation/NavigationBar" component={NavigationBar} />
+<Meta title="Overlays/Navigation/NavigationBar" component={NavigationBar} />
 
 # NavigationBar
 
diff --git a/packages/cds-web-overlays/src/navigation/__stories__/NavigationStorySetup.tsx b/packages/cds-web-overlays/src/navigation/__stories__/NavigationStorySetup.tsx
index 8c67250ff..e580a5c32 100644
--- a/packages/cds-web-overlays/src/navigation/__stories__/NavigationStorySetup.tsx
+++ b/packages/cds-web-overlays/src/navigation/__stories__/NavigationStorySetup.tsx
@@ -1,32 +1,26 @@
 /* eslint-disable react-perf/jsx-no-new-function-as-prop */
 import React, { useCallback, useMemo, useState } from 'react';
 import { css } from 'linaria';
+import { HStack } from '@cbhq/cds-web/alpha/HStack';
+import { VStack } from '@cbhq/cds-web/alpha/VStack';
+import { AvatarButton, IconButton } from '@cbhq/cds-web/buttons/index';
+import { LogoMark, NavigationIcon } from '@cbhq/cds-web/icons';
+import { Pictogram } from '@cbhq/cds-web/illustrations';
+import { LoremIpsum } from '@cbhq/cds-web/layout/__stories__/LoremIpsum';
+import { Avatar } from '@cbhq/cds-web/media';
+import { NavigationBar, NavigationTitle, NavLink, Sidebar } from '@cbhq/cds-web/navigation/index';
+import { navLinkClassName } from '@cbhq/cds-web/navigation/NavLink';
+import { PortalProvider } from '@cbhq/cds-web/overlays/PortalProvider';
+import { Pressable } from '@cbhq/cds-web/system';
+import { TabNavigation } from '@cbhq/cds-web/tabs';
+import { MockTabPanel } from '@cbhq/cds-web/tabs/__stories__/MockTabPanel';
+import { palette } from '@cbhq/cds-web/tokens';
+import { TextDisplay2, TextHeadline, TextLegal } from '@cbhq/cds-web/typography';
+import { enableJavascript } from '@cbhq/cds-web/utils/storybookParams/percy';
 
-import { HStack } from '../../alpha/HStack';
-import { VStack } from '../../alpha/VStack';
-import { AvatarButton, IconButton } from '../../buttons/index';
-import { SelectOption } from '../../controls/SelectOption';
-import { LogoMark, NavigationIcon } from '../../icons';
-import { Pictogram } from '../../illustrations';
-import { LoremIpsum } from '../../layout/__stories__/LoremIpsum';
-import { Avatar } from '../../media';
-import { PortalProvider } from '../../overlays/PortalProvider';
-import { Pressable } from '../../system';
-import { TabNavigation } from '../../tabs';
-import { MockTabPanel } from '../../tabs/__stories__/MockTabPanel';
-import { palette } from '../../tokens';
-import { TextDisplay2, TextHeadline, TextLegal } from '../../typography';
-import { enableJavascript } from '../../utils/storybookParams/percy';
-import {
-  NavigationBar,
-  NavigationTitle,
-  NavLink,
-  Sidebar,
-  SidebarItem,
-  SidebarItemProps,
-} from '../index';
-import { navLinkClassName } from '../NavLink';
-import { SidebarMoreMenu, SidebarMoreMenuProps } from '../SidebarMoreMenu';
+import { SelectOption } from '../../select/SelectOption';
+import { SidebarItem, SidebarItemProps } from '../../sidebarMoreMenu/SidebarItem';
+import { SidebarMoreMenu, SidebarMoreMenuProps } from '../../sidebarMoreMenu/SidebarMoreMenu';
 
 export const StoryMap = {
   NoTabsNoTitle: 'No Tabs no displayTitle',
diff --git a/packages/cds-web-overlays/src/navigation/__stories__/Sidebar.stories.mdx b/packages/cds-web-overlays/src/navigation/__stories__/Sidebar.stories.mdx
index 5469c4f98..2af32c044 100644
--- a/packages/cds-web-overlays/src/navigation/__stories__/Sidebar.stories.mdx
+++ b/packages/cds-web-overlays/src/navigation/__stories__/Sidebar.stories.mdx
@@ -1,10 +1,10 @@
 import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
-import { withStoryLabel } from '../../storybook-decorators/withStoryLabel';
+import { withStoryLabel } from '@cbhq/cds-web/storybook-decorators/withStoryLabel';
 
 import { SidebarExample } from './NavigationStorySetup';
-import { Sidebar } from '../Sidebar';
+import { Sidebar } from '@cbhq/cds-web/navigation/Sidebar';
 
-<Meta title="Core Components/Navigation/Sidebar" component={Sidebar} />
+<Meta title="Overlays/Navigation/Sidebar" component={Sidebar} />
 
 # Sidebar
 
diff --git a/packages/cds-web-overlays/src/navigation/__tests__/SidebarMoreMenu.test.tsx b/packages/cds-web-overlays/src/navigation/__tests__/SidebarMoreMenu.test.tsx
index 5dad5fe82..a11a41896 100644
--- a/packages/cds-web-overlays/src/navigation/__tests__/SidebarMoreMenu.test.tsx
+++ b/packages/cds-web-overlays/src/navigation/__tests__/SidebarMoreMenu.test.tsx
@@ -1,13 +1,13 @@
 import { useMemo } from 'react';
 import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
+import { NavigationIcon } from '@cbhq/cds-web/icons';
+import { SidebarProvider } from '@cbhq/cds-web/navigation/SidebarContext';
 import { renderA11y } from '@cbhq/cds-web-utils/jest';
 
-import { SelectOption } from '../../controls';
-import { NavigationIcon } from '../../icons';
+import { SelectOption } from '../../select/SelectOption';
+import { SidebarMoreMenu } from '../../sidebarMoreMenu/SidebarMoreMenu';
 import { items } from '../__stories__/NavigationStorySetup';
-import { SidebarProvider } from '../SidebarContext';
-import { SidebarMoreMenu } from '../SidebarMoreMenu';
 
 const TRIGGER_TEST_ID = 'sidebar-more-menu-trigger';
 const MENU_TEST_ID = 'sidebar-more-menu-menu';
