diff --git a/packages/cds-web-overlays/src/overlays/__stories__/AllOverlays.stories.tsx b/packages/cds-web-overlays/src/overlays/__stories__/AllOverlays.stories.tsx
index 619187d0a..1fd056837 100644
--- a/packages/cds-web-overlays/src/overlays/__stories__/AllOverlays.stories.tsx
+++ b/packages/cds-web-overlays/src/overlays/__stories__/AllOverlays.stories.tsx
@@ -1,20 +1,21 @@
 import React, { useCallback, useState } from 'react';
 import { loremIpsum } from '@cbhq/cds-common/internal/data/loremIpsum';
+import { Button } from '@cbhq/cds-web/buttons';
+import { VStack } from '@cbhq/cds-web/layout';
+import { Alert, Modal, ModalBody, ModalFooter, ModalHeader } from '@cbhq/cds-web/overlays';
+import { PortalProvider } from '@cbhq/cds-web/overlays/PortalProvider';
+import { useAlert } from '@cbhq/cds-web/overlays/useAlert';
+import { useModal } from '@cbhq/cds-web/overlays/useModal';
+import { useToast } from '@cbhq/cds-web/overlays/useToast';
+import { TextBody } from '@cbhq/cds-web/typography';
+import { enableJavascript } from '@cbhq/cds-web/utils/storybookParams/percy';
 
-import { Button } from '../../buttons';
-import { Select, SelectOption } from '../../controls';
-import { DropdownProps } from '../../dropdown';
 import { Default as DropdownExample } from '../../dropdown/__stories__/Dropdown.stories';
-import { VStack } from '../../layout';
+import { DropdownProps } from '../../dropdown/DropdownProps';
 import { SidebarExample } from '../../navigation/__stories__/NavigationStorySetup';
-import { TextBody } from '../../typography';
-import { enableJavascript } from '../../utils/storybookParams/percy';
-import { Alert, Modal, ModalBody, ModalFooter, ModalHeader } from '../index';
-import { PortalProvider } from '../PortalProvider';
-import { Tooltip } from '../Tooltip/Tooltip';
-import { useAlert } from '../useAlert';
-import { useModal } from '../useModal';
-import { useToast } from '../useToast';
+import { Select } from '../../select/Select';
+import { SelectOption } from '../../select/SelectOption';
+import { Tooltip } from '../../tooltip/Tooltip';
 
 const options = [
   'Option 1',
@@ -103,6 +104,6 @@ export const CompositeOverlaysWithoutPortal = () => <CompositeOverlays disablePo
 CompositeOverlaysWithoutPortal.parameters = { percy: enableJavascript };
 
 export default {
-  title: 'Core Components/Overlays',
+  title: 'Overlays/Overlays/All Overlays',
   component: CompositeOverlays,
 };
diff --git a/packages/cds-web-overlays/src/overlays/__stories__/SearchInputMenu.stories.tsx b/packages/cds-web-overlays/src/overlays/__stories__/SearchInputMenu.stories.tsx
index 3454abe43..04c4fc63f 100644
--- a/packages/cds-web-overlays/src/overlays/__stories__/SearchInputMenu.stories.tsx
+++ b/packages/cds-web-overlays/src/overlays/__stories__/SearchInputMenu.stories.tsx
@@ -1,12 +1,13 @@
 import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { SearchInput } from '@cbhq/cds-web/controls/SearchInput';
+import { Icon } from '@cbhq/cds-web/icons';
+import { HStack } from '@cbhq/cds-web/layout/HStack';
+import { Pressable } from '@cbhq/cds-web/system/Pressable';
+import { TextCaption, TextLabel1 } from '@cbhq/cds-web/typography';
 
-import { SearchInput } from '../../controls/SearchInput';
-import { SelectOption } from '../../controls/SelectOption';
-import { Dropdown, DropdownRefProps } from '../../dropdown';
-import { Icon } from '../../icons';
-import { HStack } from '../../layout/HStack';
-import { Pressable } from '../../system/Pressable';
-import { TextCaption, TextLabel1 } from '../../typography';
+import { Dropdown } from '../../dropdown/Dropdown';
+import { DropdownRefProps } from '../../dropdown/DropdownProps';
+import { SelectOption } from '../../select/SelectOption';
 
 const options = [
   'Option 1',
